pkgs/tools/security/gnupg/21.nix

```diff
@@ -23,9 +23,14 @@ stdenv.mkDerivation rec {
   };
 
   buildInputs = [
-    pkgconfig libgcrypt libassuan libksba libiconv npth gettext texinfo
-    readline libusb gnutls adns openldap zlib bzip2
-  ];
+    pkgconfig libgcrypt libassuan libksba libiconv npth gettext texinfo ]
+      ++ optional (adns     != null) adns
```

Comment:
If the buildInput is present on the system, I would assume there is not much difference. However, if you are developing on a local nixpkgs copy and you trigger the build of all the dependencies before you get to build yours - that takes a lot of time. For instance pinentry -> X11 -> mesa etc. There are many more examples of this in packages already, for instance octave (pkgs/development/interpreters/octave/default.nix).